sycl/test-e2e/DeviceDependencies/free_function_kernels.cpp

```diff
@@ -0,0 +1,270 @@
+  for (int i = 0; i < size; i++) {
+    if (usmPtr[i] != Result[i]) {
+      Pass = false;
+      break;
+    }
+  }
+  if (Pass)
+    return true;
+
+  std::cout << "Expected = {";
+  for (int i = 0; i < size; i++) {
+    std::cout << Result[i] << ", ";
+  }
+  std::cout << "}\n";
+  std::cout << "Result   = {";
+  for (int i = 0; i < size; i++) {
+    std::cout << usmPtr[i] << ", ";
+  }
+  std::cout << "}\n";
+  return false;
+}
+
+extern "C" SYCL_EXT_ONEAPI_FUNCTION_PROPERTY(
+    (ext::oneapi::experimental::single_task_kernel)) void ff_0(int *ptr,
+                                                               int start,
+                                                               int end) {
+  for (int i = start; i <= end; i++)
+    ptr[i] = start + end;
+}
+
+bool test_0(queue Queue) {
+  constexpr int Range = 10;
+  int *usmPtr = malloc_shared<int>(Range, Queue);
+  int start = 3;
+  int end = 5;
+  int Result[Range] = {0, 0, 0, 8, 8, 8, 0, 0, 0, 0};
+  range<1> R1{Range};
+
+  memset(usmPtr, 0, Range * sizeof(int));
+  Queue.submit([&](handler &Handler) {
+    Handler.single_task([=]() {
+      for (int i = start; i <= end; i++)
+        usmPtr[i] = start + end;
+    });
+  });
+  Queue.wait();
+  bool PassA = checkUSM(usmPtr, Range, Result);
+  std::cout << "Test 0a: " << (PassA ? "PASS" : "FAIL") << std::endl;
+
+  bool PassB = false;
+#ifndef __SYCL_DEVICE_ONLY__
+  kernel_bundle Bundle =
+      get_kernel_bundle<bundle_state::executable>(Queue.get_context());
+  kernel_id Kernel_id = ext::oneapi::experimental::get_kernel_id<ff_0>();
+  kernel Kernel = Bundle.get_kernel(Kernel_id);
+  memset(usmPtr, 0, Range * sizeof(int));
+  Queue.submit([&](handler &Handler) {
+    Handler.set_arg(0, usmPtr);
+    Handler.set_arg(1, start);
+    Handler.set_arg(2, end);
+    Handler.single_task(Kernel);
+  });
+  Queue.wait();
+  PassB = checkUSM(usmPtr, Range, Result);
+  std::cout << "Test 0b: " << (PassB ? "PASS" : "FAIL") << std::endl;
+
+  free(usmPtr, Queue);
+#endif
+  return PassA && PassB;
+}
+
+// Overloaded free function definition.
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY(
+    (ext::oneapi::experimental::nd_range_kernel<1>))
+void ff_1(int *ptr, int start, int end) {
+  nd_item<1> Item = ext::oneapi::this_work_item::get_nd_item<1>();
+  id<1> GId = Item.get_global_id();
+  ptr[GId.get(0)] = GId.get(0) + start + end;
+}
+
+bool test_1(queue Queue) {
+  constexpr int Range = 10;
+  int *usmPtr = malloc_shared<int>(Range, Queue);
+  int start = 3;
+  int Result[Range] = {13, 14, 15, 16, 17, 18, 19, 20, 21, 22};
+  nd_range<1> R1{{Range}, {1}};
+
+  memset(usmPtr, 0, Range * sizeof(int));
+  Queue.submit([&](handler &Handler) {
+    Handler.parallel_for(R1, [=](nd_item<1> Item) {
+      id<1> GId = Item.get_global_id();
+      usmPtr[GId.get(0)] = GId.get(0) + start + Range;
+    });
+  });
+  Queue.wait();
+  bool PassA = checkUSM(usmPtr, Range, Result);
+  std::cout << "Test 1a: " << (PassA ? "PASS" : "FAIL") << std::endl;
```

Comment:
I don't think that our E2E tests should print anything in a positive scenario when they pass. The motivation for that is to reduce amount of I/O performed by tests and less data that has to be processed by `lit` framework. Logs for passed tests are hidden anyways, so they are of no use in our CI.

Reply:
Add TODO comment

Reply:
I would rather immediately drop (almost) all `std::cout` lines, but I will leave that up to SYCL RT reviewers, who are code owners of the file